optimum/intel/openvino/modeling_base_seq2seq.py

```diff
@@ -188,16 +189,41 @@ def _from_pretrained(
 
         quantization_config = cls._prepare_weight_quantization_config(quantization_config, load_in_8bit)
 
+        compile_only = kwargs.get("compile_only", False)
+
         # Load model from a local directory
         if os.path.isdir(model_id):
-            encoder = cls.load_model(os.path.join(model_id, encoder_file_name), quantization_config)
-            decoder = cls.load_model(os.path.join(model_id, decoder_file_name), quantization_config)
-            if use_cache:
-                decoder_with_past = cls.load_model(
-                    os.path.join(model_id, decoder_with_past_file_name), quantization_config
-                )
-
             model_save_dir = Path(model_id)
+            if not compile_only:
+                encoder = cls.load_model(os.path.join(model_id, encoder_file_name), quantization_config)
+                decoder = cls.load_model(os.path.join(model_id, decoder_file_name), quantization_config)
+                if use_cache:
+                    decoder_with_past = cls.load_model(
+                        os.path.join(model_id, decoder_with_past_file_name), quantization_config
+                    )
+            else:
+                encoder = cls._compile_model(
+                    os.path.join(model_id, encoder_file_name),
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    True,
+                    model_save_dir,
+                )
+                decoder = cls._compile_model(
+                    os.path.join(model_id, decoder_file_name),
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    True,
+                    model_save_dir,
+                )
+                if use_cache:
+                    decoder_with_past = cls._compile_model(
+                        os.path.join(model_id, decoder_with_past_file_name),
+                        kwargs.get("device", "CPU"),
+                        kwargs.get("ov_config"),
+                        True,
+                        model_save_dir,
+                    )
```

Comment:
Here what about skipping loading and update the request in the `.__init__()` instead ?

Reply:
I think it will be messy... initially request was used for infer request object and some models (e.g.decoder in seq2seq and cualalm models because request gives opportunity to get raw openvino tensors without coping, it gives better memory management for nonCPU devices), others appears on compiled model ability to create infer request inside due to this misalignment  - some models will have model filed others not.

Besides that if we will keep only infer request for decoders and still plan introduce concurrency mode, we need to keep object that produces infer requests (compiled model) to be able create additional infer request even in compile only mode

Reply:
if you mean that this should be moved to `__init__`, then I think it will requires to handle all paths as arguments of `__init__`